run.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -e
+cd `dirname $0`
+
+echo "Current Directory: $(pwd)"
+echo "Checking for .installed file..."
+
+if [ -f .installed ] 
+  then
+    echo "Virtual environment found, activating..."
+    source viam-env/bin/activate
+    echo "Virtual environment activated: $VIRTUAL_ENV"
+  else
+    echo "Setting up virtual environment..."
+    python3 -m pip install --user virtualenv
+    python3 -m venv viam-env
+    source viam-env/bin/activate
+    echo "Virtual environment activated: $VIRTUAL_ENV"
+    echo "Installing dependencies from requirements.txt..."
+
+    while IFS= read -r requirement; do
+      if [ -n "$requirement" ]; then
+        echo "Installing $requirement..."
+        pip install "$requirement" || echo "Failed to install $requirement"
+      fi
+    done < requirements.txt
```

Comment:
IMO this seems like it could be too verbose for what it's worth. Could we just do `pip install -r requirements.txt`?

Reply:
I had that at first, but then I ran into the problem of specific requirements failing to build and then after that none of the requirements underneath it in the list were installed because it just exited from the requirements.txt file entirely. This was really annoying to me so that is why I added in this loop - but I can definitely take it out.